src/newlines.cpp

```diff
@@ -2734,7 +2735,8 @@ static void newline_before_return(Chunk *start)
 
    Chunk *pc = Chunk::NullChunkPtr;
 
-   if (start != nullptr)
+   if (  start != nullptr
+      && start->IsNotNullChunk())
```

Comment:
``` if (start != nullptr)``` was added to prevent SEGV during the transition process of the code improvement. At the end this will simply be ```pc = start->GetPrev();``` without any ```if```, because Chunk pointers won't be null.
Checking for ```start->IsNotNullChunk()``` is not wrong, but it is not necessary either. If ```start ``` is a null chunk, ```start ->GetPrev()``` will return a Chunk::NullChunkPtr, with no effect on the current value of ```pc```